cvpaperchallenge/Crux
# 5/enhancement/implement frontend container

## Issue URL

#5 

## Change overview

- Add necessary python packages for streamlit to `pyproject.toml` and `poetry.lock`
- Implement a simple web page using streamlit api
- Make a Dockerfile to construct a Docker container containing files under `applications/frontend`

## How to test

1. Run the following command under project root to build a Docker image
`docker build -t crux-frontend-debug -f environments/frontend_Dockerfile .` 
2. Run the following command consecutively to construct a Docker container
 `docker run -it -p 8501:8501 -v /home/botany/projects/Crux/applications/frontend/:/home/challenger/crux-frontend --name crux-frontend-debug crux-frontend-debug`
3. Inside the container, run `poetry install`  to install python packages necessary for streamlit
4. Run the following command to host the local server
`poetry run streamlit run src/test_streamlit.py --server.port 8501 --server.address 0.0.0.0`
5. You can verify that it works properly by visiting the server and confirming the message "Welcome to Crux!"

## Note for reviewers

close #5 
If both #5 and #6 have been completed, we can move on to work on #4.
